easycv/transforms/detect.py

```diff
@@ -44,6 +47,123 @@ def process(self, image, **kwargs):
         return {"detections": len(decoded), "data": data, "rectangles": rectangles}
 
 
+class CascadeDetector(Transform):
+
+    arguments = {
+        "cascade": File(),
+        "scale": Number(default=1.1),
+        "min_neighbors": Number(min_value=0, only_integer=True, default=3),
+        "min_size": Number(min_value=0, default="auto"),
+        "max_size": Number(min_value=0, default="auto"),
+    }
+
+    outputs = {
+        "rectangles": List(
+            List(List(Number(min_value=0, only_integer=True), length=2), length=2)
+        ),
+    }
+
+    def process(self, image, **kwargs):
+        cascade = cv2.CascadeClassifier(kwargs["cascade"])
+        gray = GrayScale().apply(image)
+        detections = cascade.detectMultiScale(
+            gray,
+            scaleFactor=kwargs["scale"],
+            minNeighbors=kwargs["min_neighbors"],
+            minSize=kwargs["min_size"] if kwargs["min_size"] != "auto" else None,
+            maxSize=kwargs["max_size"] if kwargs["max_size"] != "auto" else None,
+        )
+
+        rectangles = []
+        for x, y, w, h in detections:
+            rectangles.append([(x, y), (x + w, y + h)])
+
+        return {"rectangles": rectangles}
+
+
+class Faces(Transform):
+
+    arguments = {
+        "scale": Number(default=1.3),
+        "min_neighbors": Number(min_value=0, only_integer=True, default=5),
+    }
+
+    outputs = {
+        "rectangles": List(
+            List(List(Number(min_value=0, only_integer=True), length=2), length=2)
+        ),
+    }
+
+    def process(self, image, **kwargs):
+        cascade_file = get_resource(
+            "haar-face-cascade", "haarcascade_frontalface_default.xml"
+        )
+        return CascadeDetector(cascade=str(cascade_file), **kwargs).apply(image)
+
+
+class Eyes(Transform):
+
+    arguments = {
+        "scale": Number(default=1.1),
+        "min_neighbors": Number(min_value=0, only_integer=True, default=3),
+    }
+
+    outputs = {
+        "rectangles": List(
+            List(List(Number(min_value=0, only_integer=True), length=2), length=2)
+        ),
+    }
+
+    def process(self, image, **kwargs):
+        cascade_file = get_resource("haar-eye-cascade", "haarcascade_eye.xml")
+
+        rectangles = []
+        for face in Faces().apply(image)["rectangles"]:
+            face_image = Crop(rectangle=face).apply(image)
+            eyes = CascadeDetector(cascade=str(cascade_file), **kwargs).apply(
+                face_image
+            )["rectangles"]
+            for eye in eyes:
+                adjusted = []
+                for i in range(len(eye)):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/easycv/easycv/pullRequest?prid=5694061)